### PR TITLE
Fix the copy-assets script to support an absolute path to m-c, and su…

### PR DESCRIPTION
**Pull request by julienw**
…pport symlinking to an outoftree m-c clone

### CODE REVIEW

**Comment by julienw** [comment]
I tested with an outoftree m-c clone because that was my issue. I'd appreciate if the reviewer could test an in-tree clone (I mean, the way recommended in the docs).

File: bin/copy-assets.js [hunk 1]
```diff
@@ -19,50 +19,52 @@ function start() {
   console.log("start: copy assets")
   const projectPath = path.resolve(__dirname, "..")
   const mcModulePath =  "devtools/client/debugger/new";
-  const mcPath = args.mc ? args.mc : feature.getValue("firefox.mcPath");
+  let mcPath = args.mc ? args.mc : feature.getValue("firefox.mcPath");
+
+  // resolving against the project path in case it's relative. If it's absolute
+  // it will override whatever is in projectPath.
+  mcPath = path.resolve(projectPath, mcPath);
```
Comment (julienw) [inline bin/copy-assets.js]: This is the magic.

File: bin/copy-assets.js [hunk 2]
```diff
@@ -19,50 +19,52 @@ function start() {
...
-    path.join(projectPath, mcPath, "devtools/client/debugger/new/moz.build"),
+    path.join(mcPath, "devtools/client/debugger/new/moz.build"),
     {cwd: projectPath}
   );
 
+  const projectTestPath = path.join(projectPath, "src/test/mochitest");
+  const mcTestPath = path.join(mcPath, mcModulePath, "test/mochitest");
   if (shouldSymLink) {
-    symlinkTests({ projectPath, mcModulePath })
+    symlinkTests({ projectPath, mcTestPath, projectTestPath })
```
Comment (julienw) [inline bin/copy-assets.js]: I've seen symlinkTests already support this, we just need to use it.

File: bin/copy-assets.js (hunk 1 above)
Comment (juliandescottes) [inline bin/copy-assets.js]: projectPath is always set as the working directory in the rest of the script, so I'm not sure if this is really needed.
I tried commenting out this line and the script still ran correctly for both in and out of tree mc clones. Am I missing something? (in any case it doesn't hurt to have it)

Comment (julienw) [inline bin/copy-assets.js]: ah you're right; except maybe the outputPath at the end ?

**Pull request merged**